docs/concepts/glossary.md

```diff
@@ -0,0 +1,166 @@
+
+A Flink Application Cluster is a dedicated [Flink Cluster](./glossary#flink-cluster) that only
+executes a single [Flink Job](./glossary#flink-job). The lifetime of the
+[Flink Cluster](./glossary#flink-cluster) is bound to the lifetime of the Flink Job. Formerly
+Flink Application Clusters were also known as Flink Clusters in *job mode*. Compare to
+[Flink Session Cluster](./glossary#flink-session-cluster).
+
+#### Flink Cluster
+
+The distributed system consisting of (typically) one Flink Master process and one or more Flink
+Taskmanagers processes.
+
+#### Event
+
+An event is a statement about a change of the state of the domain modelled by the
+application. Events can be input and/or output of a stream or batch processing application.
+Events are special types of [records](./glossary#Record)
+
+#### ExecutionGraph
+
+see [Physical Graph](./glossary#physical-graph)
+
+#### Function
+
+Functions, or user-defined functions (UDFs), are implemented by the user and encapsulate the
+application logic of a Flink program. Most Functions are wrapped by a corresponding
+[Operator](./glossary#operator).
+
+#### Instance
+
+The term *instance* is used to describe a specific instance of a specific type (usually
+[Operator](./glossary#operator) or [Function](./glossary#function)) during runtime. As Apache Flink
+is mostly written in Java, this corresponds to the definition of *Instance* or *Object* in Java.
+In the context of Apache Flink, the term *parallel instance* is also frequently used to emphasize
+that multiple instances of the same [Operator](./glossary#operator) or
+[Function](./glossary#function) type are running in parallel.
+
+#### Flink Job
+
+A Flink Job is the runtime representation of a Flink program. A Flink Job can either be submitted
+to a long running [Flink Session Cluster](./glossary#flink-session-cluster) or it can be started as a
+self-contained [Flink Application Cluster](./glossary#flink-application-cluster).
+
+#### JobGraph
+
+see [Logical Graph](./glossary#logical-graph)
+
+#### Flink JobManager
+
+JobManagers are one of the components running in the
+[Flink JobManger Process](./glossary#flink-jobmanager-process). A JobManager is responsible for
+supervising the execution of the [Tasks](./glossary#task) of a single job.
+
+#### Logical Graph
+
+A logical graph is a directed graph describing the high-level logic of a stream processing program.
+The nodes are [Operators](./glossary#operator) and the edges indicate input/output-relationships or
+data streams or data sets.
+
+#### Managed State
+
+Managed State describes application state which has been registered with the framework. For
+Managed State, Apache Flink will take care about persistence and rescaling among other things.
+
+#### Flink JobManager Process
+
+The Job Manager Process is the master of a [Flink Cluster](./glossary#flink-cluster). It is called
+*JobManager* for historical reasons, but actually has actually contains three distinct components:
```

Comment:
Currently, they are always run together, correct? I assume the other option would be: 

* Flink Master Process: Dispatcher &  Resource Manager
* One JobManager Process per JobManager separately

How about we it like it is right now and revisit this, when we have decided about introducing the term "Flink Master Process" in the course of the FLIP?

Reply:
SGTM.

Reply:
AFAIK, they are not always run together. At least that's what is described by FLIP-6. But I don't want to nitpick on this

Reply:
@fhueske In which case are these not run together? AFAIK also in YARN/Mesos session mode, there is only one Master Process for all Job Managers, although conceptually they would not have to run together (since FLIP-6).

Reply:
I talked to @tillrohrmann offline and they are currently always run together in the same process + there are no plans to change this in the foreseeable future.